### DISCUSSION TITLE
reset() function doesn't fire the onChange event

### DISCUSSION THREAD

**Discussion by pedro-olavo** [react-hook-form/react-hook-form#9326] [Q&A]
Let's say that I have this component writed below:
```
const handleOnChange (e) => {
  if (!e.currentTarget.value) {
    // do something
  }
} 
return <input onChange={handleOnChange} />
````
When I reset the form with reset(), the onChange event isn't triggered.

How can I capture that the input has been reseted?

**Comment by leapful** [comment]
@SrOlaff 

Method `reset` will not trigger `onChange` callback as there is no interaction on field element that trigger change when re-setting value.

You can use `useWatch` hook to keep track on change of form values.
